### ISSUE TITLE
Dice Score

### ISSUE THREAD

**Issue by TheCarmelGuy** [shreyaspadhy/UNet-Zoo#9]
Just wanted to make sure I'm looking at this correctly: the Average Dice Coefficient which is printed out after you run test should be different then the DICE score which is shown on the  'dice.png'

**Comment by shreyaspadhy** [comment]
Yes, that's correct! The average DICE score is printed over the batch that's currently being trained over. However, the DICE Score on 'dice.png' is calculated for that particular image.

**Comment by TheCarmelGuy** [comment]
I guess my question is this: A low Dice 'loss" reflects a better forward propagation as a smaller loss indicates you don't have to adjust your parameters during training. However, for Dice "Score", two samples with good overlap (ie. segmentation map closely matches ground truth) should technically be higher(eg. 1 for max similarity, 0 for minimum similarity). Is that a true statement?  If that is the case, then the average DICE "loss" is printed out over the training samples where as the Dice "Score" would be in the 'dice.png' file.

**Comment by shreyaspadhy** [comment]
You got it right! Given two images, a DICE score of 1 means perfect overlap, and a DICE score of 0 means no overlap at all. However, from `losses.py`, you can see the DICE Loss is calculated as `loss = 1 - torch.sum(dice_eso) / dice_eso.size(0)`, which is 1 minus the average DICE Score across all images in the batch.
